### ISSUE TITLE
Can I contribute if I am not a programmer?

### ISSUE THREAD

**Issue by orschiro** [espeak-ng/espeak-ng#291]
My apologies if this is the wrong place to ask this question. 

Basically, can I – and if how – contribute to this great project if I am not a programmer?

Can I read something aloud, for instance, that can help improve espeak?

Thanks!

**Comment by valdisvi** [comment]
Thanks for your interest!

Anybody can contribute to pronunciation rules for language, which he or she knows (e.g. German, Polish, English etc.). You can submit just issues about incorrect pronunciation in its phonetic notation,
e.g. "car" should be said as "k'A@".

But, better would be considering translation rules, which are described in [dictionary.md](https://github.com/espeak-ng/espeak-ng/blob/master/docs/dictionary.md) file. Configuration files for e.g. English are: [en_rules](https://github.com/espeak-ng/espeak-ng/blob/master/dictsource/en_rules) and [en_list](https://github.com/espeak-ng/espeak-ng/blob/master/dictsource/en_list). You can test quite recent development version of [espeak-ng online](http://odo.lv/Espeak) page.

There are some open issues about [documentation](https://github.com/espeak-ng/espeak-ng/labels/languages%2Fpronunciation) and [languages](https://github.com/espeak-ng/espeak-ng/labels/languages%2Fpronunciation), but these are quite technical and require understanding of espeak-ng infrastructure.

eSpeakNG uses [formant synthesis](https://en.wikipedia.org/wiki/ESpeakNG#Synthesis_method) to produce sound, therefore recorded voice can be used only to get voice spectrum, which then can be approximately emulated by eSpeakNG. There are other open source synthesizers, which use recorded voice, e.g. [Festival](http://www.cstr.ed.ac.uk/projects/festival/), but also for these, making of recordings is just first step in data analysis process of particular language and voice.

eSpeakNG requires specific set of technical skills to improve it, therefore list of its contributors is not long. But, if there would be considerable interest, we could create e.g. Patreon page to pay money for its contributors.

**Comment by orschiro** [comment]
Thanks for your intro! This is all very new to me. 

Let's start from the end with a concrete example.

I want to improve the narrate feature of Firefox which relies on espeak. I want the voice to sound more pleasant and more human to listen to. 

Where do I start to make this better? Or do I need to be a programmer for this?

![image](https://user-images.githubusercontent.com/170947/28748304-8a148286-74b4-11e7-8aa0-b089b84fe87c.png)

**Comment by valdisvi** [comment]
I don't know about Firefox plugin, but by default espeak-ng is shipped with few included MBROLA voices already (assuming you are using Linux, I don't know about Windows). E.g. you can make espeak-ng  talk as more natural US female MBROLA voice by entering command:
```
espeak -v mb-us1 "Hello world. This is mbrola speaking."
```
... or as UK male:
```
espeak -v mb-en1 "Hello world. This is mbrola speaking."
```
And you can search more MBROLA voices, by searching for **mbrola** packages (e.g. for Debian-like distributions):

```
apt-cache search mbrola
```
Then install e.g. ```sudo apt install mbrola-af1``` and call it passing ```mb-af1``` as voice parameter.

...but I myself got used to generated voice and I find it more recognizable than MBROLA voices, especially with higher speeds. (Mainly because of most MBROLA voices has only up to 4kHz of sound spectrum produced, but eSpeakNG has up to 11kHz).

**Comment by orschiro** [comment]
Hey, thanks! Slowly beginning to better understand this universe. I am on Solus Linux which doesn't provide the mbrola package. Will have to see how to proceed...

Other question: what's the difference between espeak and espeak-ng?

Or is -ng an alias to espeak?

![image](https://user-images.githubusercontent.com/170947/28762899-52045978-75ba-11e7-8634-e30a56e5724c.png)

**Comment by valdisvi** [comment]
Yes. Actual executable for eSpeak NG is called `espeak-ng`, and for compatibility reasons it also has symbolic link named just `espeak`.
(You will get this if you will install espeak-ng from source code, but if you will install binary package from repository, it depends how package maintainers decided to deal with this.)

**Comment by orschiro** [comment]
Thanks! Coming back to one point:

> eSpeakNG requires specific set of technical skills to improve it, therefore list of its contributors is not long. But, if there would be considerable interest, we could create e.g. Patreon page to pay money for its contributors.

I think this is an excellent idea! I feel it's quite hard for me to contribute here but would love to give you a pledge to help espeak on Linux. :-)
